### PR TITLE
chunk space uploads

### PR DESCRIPTION
**Pull request by pngwn**
## Description

Please include a concise summary, in clear English, of the changes in this pull request. If it closes an issue, please mention it here.

Closes: #(issue)

## 🎯 PRs Should Target Issues

Before your create a PR, please check to see if there is [an existing issue](https://github.com/gradio-app/gradio/issues) for this change. If not, please create an issue before you create this PR, unless the fix is very small. 

Not adhering to this guideline will result in the PR being closed. 

## Tests

1. PRs will only be merged if tests pass on CI. To run the tests locally, please set up [your Gradio environment locally](https://github.com/gradio-app/gradio/blob/main/CONTRIBUTING.md) and run the tests: `bash scripts/run_all_tests.sh`

2. You may need to run the linters: `bash scripts/format_backend.sh` and `bash scripts/format_frontend.sh`

### CODE REVIEW

File: scripts/upload_demo_to_space.py [hunk 1]
```diff
@@ -1,78 +1,159 @@
...
+                    repo_id=space_id,
+                    operations=chunk,
+                    commit_message=commit_message,
+                    token=hf_token,
+                    repo_type="space",
+                )
+                print(f"Successfully committed chunk {idx} with {len(chunk)} file(s).")
+            except Exception as e:
+                print(f"Failed to commit chunk {idx}: {e}")
+                raise e  # Re-raise the exception after logging
```
Comment (Wauplin) [inline scripts/upload_demo_to_space.py:127]:
```suggestion
                raise  # Re-raise the exception after logging
```

(nit) see https://stackoverflow.com/a/45986945

File: scripts/upload_demo_to_space.py [hunk 2]
```diff
@@ -1,78 +1,159 @@
...
+            """
+            readme.write_text(textwrap.dedent(readme_content))
+
+        print("Uploading files to Hugging Face Space...")
+        api = HfApi()
+
+        print("Creating repository...")
+
+        # Create the repository if it doesn't exist
+        api.create_repo(
```
Comment (Wauplin) [inline scripts/upload_demo_to_space.py:84]: (nit) Here you could do `space_url = api.create_repo(...)` and then `space_id = space_url.repo_id`. This way the script will also work with implicit repo_id when the username is not provided. If you move the create_repo call a bit up, you can reuse `space_id` (the correct one with explicit namespace) in the logs + model card.

**Review by Wauplin — APPROVED** [review]
Looks good to me! This should avoid any timeout now. I left minor comments but feel free to ignore.

File: scripts/upload_demo_to_space.py (hunk 2 above)
Comment (pngwn) [reply scripts/upload_demo_to_space.py:84]: Nice. Ty.

File: scripts/upload_demo_to_space.py [hunk 3]
```diff
@@ -1,78 +1,159 @@
...
+        hf_token (str): Hugging Face API token with write permissions to the space.
+        gradio_version (Optional[str]): If provided, sets the Gradio version in the created space.
+
+    Returns:
+        str: URL of the uploaded Hugging Face Space.
     """
 
+    print(f"Uploading demo '{demo_name}' to space '{space_id}'...")
+
+    def split_into_chunks(lst: List, n: int) -> List[List]:
```
Comment (abidlabs) [inline scripts/upload_demo_to_space.py:30]:
```suggestion
    def split_into_chunks(lst: list, n: int) -> list[list]:
```

File: scripts/upload_demo_to_space.py [hunk 4]
```diff
@@ -1,78 +1,159 @@
...
+        """
+        Splits a list into chunks of size n.
+
+        Args:
+            lst (List): The list to split.
+            n (int): The size of each chunk.
+
+        Returns:
+            List[List]: A list of chunks.
```
Comment (abidlabs) [inline scripts/upload_demo_to_space.py:39]: nit: unnecessary docstring as what this method does is quite clear from the name and code

File: scripts/upload_demo_to_space.py [hunk 5]
```diff
@@ -1,78 +1,159 @@
...
+        for i in range(0, len(lst), n):
+            yield lst[i : i + n]
+
     with tempfile.TemporaryDirectory() as tmpdir:
-        demo_path = pathlib.Path(pathlib.Path().absolute(), f"demo/{demo_name}")
+        demo_path = pathlib.Path.cwd() / "demo" / demo_name
+        if not demo_path.exists():
+            raise FileNotFoundError(f"Demo path '{demo_path}' does not exist.")
+
+        # Copy demo directory to temporary directory
```
Comment (abidlabs) [inline scripts/upload_demo_to_space.py:49]: Unnecessary comments, would be better to remove

File: scripts/upload_demo_to_space.py [hunk 6]
```diff
@@ -1,78 +1,159 @@
...
         shutil.copytree(demo_path, tmpdir, dirs_exist_ok=True)
+
+        # Rename 'run.py' to 'app.py'
         app_file = pathlib.Path(tmpdir, "run.py")
-        # Rename the app file to be app.py
-        app_file.rename(app_file.with_stem("app"))
+        if app_file.exists():
+            app_file.rename(app_file.with_stem("app"))
```
Comment (abidlabs) [inline scripts/upload_demo_to_space.py:55]: If you want, you could leave the original files as is, and then set `app_file: run.py` in the README below

**Review by abidlabs — APPROVED** [review]
Haven't tested since I'm traveling, but code changes look good. I left a few stylistic suggestions & if the code runs correctly, then should be good to merge!

**Comment by pngwn** [comment]
Will fix those issues thank @abidlabs. Pretty sure it works as i have tested it locally with my own API ket.

**Comment by pngwn** [comment]
Addressed everything, gonna merge and test it out.

Thanks for the reviews @Wauplin @abidlabs!

**Pull request merged**